### PR TITLE
Implement MVP for main view

### PR DESCRIPTION
**Pull request by CharliePoole**
Part of issue #95

### CODE REVIEW

**Comment by CharliePoole** [comment]
The initial commit is a partial implementation. The Presenter references the view directly rather than threw an interface. Next commit will add switch to use of the interface.

**Comment by CharliePoole** [comment]
Second commit removes two classes that have long been present in the (V2) Gui but are not actually used. One is a version of the settings dialog that uses tabs rather than a tree to organize settings. I don't find that helpful once you have more than half a dozen settings pages in the dialog so I removed it as well as the conditional code to use it. The other is SimpleSettingsDialog, which displays a single settings page by itself. That isn't used and can be recreated easily enough if we need it.

In general, I'm moving to eliminate anything we don't seem to have an immediate use for because it slows down the process of improving the code and adding features.

**Comment by CharliePoole** [comment]
Second commit removes two classes that have long been present in the (V2) Gui but are not actually used. One is a version of the settings dialog that uses tabs rather than a tree to organize settings. I don't find that helpful once you have more than half a dozen settings pages in the dialog so I removed it as well as the conditional code to use it. The other is SimpleSettingsDialog, which displays a single settings page by itself. That isn't used and can be recreated easily enough if we need it.

In general, I'm moving to eliminate anything we don't seem to have an immediate use for because it slows down the process of improving the code and adding features.

**Comment by CharliePoole** [comment]
This is complicated enough that I'm doing it in smaller chunks of commits than I originally planned. The latest handles popup processing (File, Runtimes, Recent Files) in the presenter adding some further support code from the experimental GUI in order to do it.

File: src/TestCentric/testcentric.gui/AppEntry.cs [hunk 1]
```diff
@@ -84,19 +86,20 @@ public static int Main(string[] args)
             ITestModel model = new TestModel(testEngine);
 
             log.Info("Constructing Form");
-            TestCentricMainForm form = new TestCentricMainForm(model, options);
+            TestCentricMainView view = new TestCentricMainView(model);
```
Comment (CharliePoole) [inline src/TestCentric/testcentric.gui/AppEntry.cs]: Getting options out of the view was an important step. Model still has to go but is harder.

File: src/TestCentric/testcentric.gui/Controls/TestTree.cs [hunk 2]
```diff
@@ -23,14 +23,13 @@
...
 using System.Collections.Generic;
-using System.ComponentModel;
 using System.Windows.Forms;
 using NUnit.Engine;
 
 namespace TestCentric.Gui.Controls
 {
     using Model;
-    using Model.Settings;
+    using Presenters;
```
Comment (CharliePoole) [inline src/TestCentric/testcentric.gui/Controls/TestTree.cs]: Undesirable, but needed for the moment.

**Comment by CharliePoole** [comment]
With the latest commit, the view no longer references the presenter or the gui options. It doesn't save a reference to the model or respond to any model events but it does hold a reference to the user settings, which are part of the model, and still updates some of them directly. It also uses some model Types, like TestNode.

With all this refactoring, things are still working! I keep running into small bugs, which were there already, and leave them in place so as to focus on the task at hand. There should be lots of smaller things to fix once this restructuring is complete.

**Comment by CharliePoole** [comment]
This one has been a slog. With the last commit, all references from the main view to the model are gone, so we are properly doing MVP. There's still cleanup to do and the tests need to be added back in.

BTW, I removed the tests because they depended on being able to create a view using a default constructor, which hasn't been possible until this point. Some temporary tests might have been a better idea, but that's hindsight.

**Comment by CharliePoole** [comment]
Aside: this commit also marks my full conversion to Linux. It was completed under Pop OS 18.04 on my new System76 laptop, which will be accompanying me on my year in Europe.

**Comment by CharliePoole** [comment]
This is now ready for reviewing and merging. 

There are many many changes and I'm embarrassed to ask for reviews, but there you go!

Rather than looking at every detail, I suggest looking at changes to the main view and presenter primarily, looking at other code if necessary to understand. If it's too much for you to look at, let me know and I'll merge it and take my chances.

FWIW, I'm going back to smaller diffs for future PRs! Remaining items are the `TestTree` and `TestSuiteTreeView` views, which now look a bit easier than they did before I made the changes here.

**Comment by CharliePoole** [comment]
Merging... I'll still try to incorporate any comments in the next PR.

**Comment by immeraufdemhund** [comment]
sorry Charlie, I mean to look at this a day  or two ago, but got held up with work

**Comment by CharliePoole** [comment]
Still time if you want to. I noticed last night that it won't merge automatically.

**Comment by immeraufdemhund** [comment]
wowza. Yeah i'm looking at gitk and I think it might be easier if you were to rebase on top of master. That would give you an opportunity to replace the commit message WIP in e7abfd6ac23b271a87ae6de7e0fd1e1bad1a1cc7. I'll start looking at the commits though.

**Comment by mikkelbu** [comment]
Sorry about the silence, but I've been quiet busy at work the last couple of weeks, so I've had very little free time to do any NUnit work.

**Comment by CharliePoole** [comment]
No problem. I understand that you've been busy. If you have any time at the moment, feel free to look through issues and suggest any to me that seem suitable for Hacktoberfest!

**Pull request merged**